pkg/service/repair/controller.go

```diff
@@ -177,50 +172,39 @@ func (c *rowLevelRepairController) TryBlock(hosts []string) (bool, allowance) {
 	return true, a
 }
 
+func (c *rowLevelRepairController) parallelLimit() int {
+	p := c.intensity.Parallel()
+
+	if p <= defaultParallel {
+		return c.intensity.MaxParallel()
+	}
+
+	return p
+}
+
 func (c *rowLevelRepairController) shouldBlock(hosts []string, intensity float64) bool {
 	// ALLOW if nothing is running
 	if !c.Busy() {
 		return true
 	}
 
-	// DENY if any host has max nr of jobs already running
+	// DENY if any host has a job running already. We don't allow more than a single repair job per host.
 	for _, h := range hosts {
-		if c.jobs[h] >= c.limits[h].Default {
+		if c.jobs[h] >= 1 {
 			return false
 		}
 	}
 
-	// DENY if any host already runs at the full intensity.
-	// This may happen if we decrease intensity, for a while we need to reduce
-	// number of jobs to reduce number of busy ranges.
-	ranges := c.rangesForIntensity(hosts, intensity)
-	for _, h := range hosts {
-		var (
-			r  = c.busyRanges[h] + ranges
-			ok bool
-		)
-		switch {
-		case intensity == maxIntensity:
-			ok = r <= c.limits[h].Max
-		case intensity <= 1:
-			ok = r <= c.limits[h].Default
-		default:
-			ok = r <= int(intensity)*c.limits[h].Default && r <= c.limits[h].Max
-		}
-		if !ok {
-			return false
-		}
+	// DENY (Sanity) if allJobs + 1 > MaxParallel.
+	// However, this should never happen due to "up to 1 repair task per replica" and the fact that MaxParallel is
+	// calculated the way that allows up to 1 repair task per host.
+	if c.allJobs+1 > c.intensity.MaxParallel() {
+		return false
```

Comment:
Btw, it seems the function returns a reversed state. shouldBlock() returns false to perform the  block? Why?

Reply:
This wasn't my decision - I'm just fixing what was written before me ;)
But I agree, some functions names do not match their semantics in a straight forward way.

But logic wise the code does what it's supposed to: https://github.com/scylladb/scylla-manager/blob/master/pkg/service/repair/controller.go#L49

Reply:
Yes, it is existing issue and is outside the scope of the PR.